sspl_test/functional_tests/alerts/node/test_node_bmc_interface.py

```diff
@@ -0,0 +1,99 @@
+
+from sspl_constants import BMCInterface
+from alerts.node import simulate_bmc_interface_alert
+from framework.base.functional_test_base import TestCaseBase
+
+from cortx.utils.process import SimpleProcess
+
+
+class BmcInterfaceTest(TestCaseBase):
+    def init(self):
+        pass
+
+    def filter(self, msg):
+        try:
+            # Make sure we get back the message type that matches the request
+            msg_type = msg.get("sensor_response_type")
+            if (
+                msg_type["info"]["resource_type"] == "node:bmc:interface:kcs"
+                or msg_type["info"]["resource_type"] == "node:bmc:interface:rmcp"
+            ):
+                return True
+        except Exception as exception:
+            print(exception)
+
+    def request(self):
+        # backup active bmc interface
+        BMC_IF_CONSUL_KEY, BMC_IF_CONSUL_VAL = self.backup_bmc_config()
+
+        if BMC_IF_CONSUL_VAL == "lan":
+            simulate_bmc_interface_alert.lan_channel_alert(
+                BMC_IF_CONSUL_KEY, BMC_IF_CONSUL_VAL
+            )
+        else:
+            simulate_bmc_interface_alert.kcs_channel_alert(
+                BMC_IF_CONSUL_KEY, BMC_IF_CONSUL_VAL
+            )
+
+    def response(self, msg):
+        bmc_interface_message = msg.get("sensor_response_type")
+
+        assert bmc_interface_message is not None
+        assert bmc_interface_message.get("alert_type") is not None
+        alert_type = bmc_interface_message.get("alert_type")
+        assert alert_type == "fault"
+        assert bmc_interface_message.get("alert_id") is not None
+        assert bmc_interface_message.get("severity") is not None
+        assert bmc_interface_message.get("host_id") is not None
+        assert bmc_interface_message.get("info") is not None
+
+        bmc_interface_info = bmc_interface_message.get("info")
+        assert bmc_interface_info.get("site_id") is not None
+        assert bmc_interface_info.get("rack_id") is not None
+        assert bmc_interface_info.get("node_id") is not None
+        assert bmc_interface_info.get("cluster_id") is not None
+        assert bmc_interface_info.get("resource_id") is not None
+        assert bmc_interface_info.get("description") is not None
+
+        bmc_interface_specific_info = bmc_interface_message.get("specific_info")
+        if bmc_interface_specific_info:
+            assert bmc_interface_specific_info.get("channel info") is not None
+
+    def backup_bmc_config(self):
+        path = BMCInterface.ACTIVE_BMC_IF.value
+        cmd = f"cat {path}"
+        bmc_interface, error, retcode = SimpleProcess(cmd).run()
+        if retcode != 0:
+            print(f"command:{cmd} not executed successfully")
+            return
+
+        # bmc_interface = b'\x80\x03X\x06\x00\x00\x00systemq\x00.\n'
+        # fetch interface key and value from bmc_interface
+        active_bmc_IF_key = path
+        # parse string b'\x80\x03X\x06\x00\x00\x00systemq\x00.\n' to fetch bmc interface value
```

Comment:
whats the hex code showing up in comments. I think its pickled data being referred directly, it should be unickled and used